arai60/83-remove-duplicates-from-sorted-list/step2.py

```diff
@@ -0,0 +1,26 @@
+class SolutionWithDoubledLoop:
+    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        node = head
+        while node is not None:
+            while node.next is not None and node.val == node.next.val:
+                node.next = node.next.next
+            node = node.next
+        return head
+
+class SolutionWithANewLinkedList:
+    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        if head is None:
+            return None
+        node = head
+        new_head = ListNode(head.val)
+        new_node = new_head
+
+        while node is not None and node.next is not None:
+            if node.val == node.next.val:
+                node.next = node.next.next
+                continue
+            node = node.next
+            added_node = ListNode(node.val)
+            new_node.next = added_node
+            new_node = added_node
```

Comment:
`added_node`という変数を定義せずに単に以下のようにも書けるかなと思いました

```suggestion
            new_node.next = ListNode(node.val)
            new_node = new_node.next
```

Reply:
修正しました！提示頂いたやり方の方が、node = node.next と対比して書くことができるのでわかりやすいですね。
https://github.com/kazizi55/coding-challenges/pull/3/files#diff-153271ae97af52ab3609bb5e1cd3215deb7bc0ef64f6750b2220ed75868283f4R1-R14